lib/testscript_engine/assertion.rb

```diff
@@ -153,13 +153,23 @@ def header_field(assert)
     compare("Header #{assert.headerField}", received, assert.operator, expected)
   end
 
+  def deep_merge(first, second)
+    merger = proc { |_, v1, v2| Hash === v1 && Hash === v2 ? v1.merge(v2, &merger) : Array === v1 && Array === v2 ? v1 | v2 : [:undefined, nil, :nil].include?(v2) ? v1 : v2 }
+    return first.merge(second.to_h, &merger)
+  end
+
   def minimum_id(assert)
     received = get_resource(assert.sourceId)
+    minimum = get_resource(assert.minimumId)
 
-    raise AssertionException.new('minimumId assert not yet supported.', :skip)
-    # result = client.validate(received, { profile_uri: assert.validateProfileId })
+    if deep_merge(received.to_hash, minimum.to_hash) == received.to_hash
```

Comment:
> let's include the check (`== received.to_hash`) within the `deep_merge` method. I think the `deep_merge` method is highly specific to this use case, so we can rename it to something like "check hash equivalency" or "check resourceA in resourceB". Probably better naming than what I'm suggesting. So, have that method do the transforming to hash and the check that the first within the second.

I agree. It is consistent to have all assert with the same compare pattern. One minor issue is it's tricky to create a success/failure message with both minimum_id and validateProfileId given the message composing method. This may be a separate issue.

Reply:
> 

That's good idea!